Python.md

```diff
@@ -40,7 +85,37 @@ result = connection.execute(sqlalchemy.text("""
            show_gravatar
       FROM "user"
      WHERE musicbrainz_id IN :musicbrainz_usernames
-"""), {
-    'musicbrainz_usernames': tuple(usernames),
-})
+""")
+result = connection.execute(query, {'musicbrainz_usernames': tuple(usernames)})
 ```
+
+It is preferred to assign the query to a separate variable and not include it inside the
+`execute` call.
+
+TODO: Projects which don't use sqlalchemy
+
+### Tests
+
+You should write tests for all new methods that you write. If you update a method, make sure that
+tests for that method still pass, and write/remove tests as necessary for the change in functionality.
+
+Tests for a module (`package/module.py`) go in a `test` subdirectory - `package/test/test_module.py`
+
+Database tests (for modules in `db`) can use the database. Inherit from `db.testing.DatabaseTestCase`
+to automatically reset the database at the end of each testcase.
```

Comment:
Right, perhaps we can clarify this a bit: 
 * Tests which use the database should pull down and recreate the database schema each time they run - each project may have a different way of doing this
 * database tests may not be in the `db` module.

Again, you know that this documentation came directly from my AB PR. A differing opinion is useful to make these docs more understandable by everyone, but you're probably in the unique position to know exactly how this works differently in all MeB flask projects (I definitely don't). Specific recommendations of text to add here would be appreciated.